test/runners/rust_spec.js

```diff
@@ -103,165 +113,352 @@ describe('rust runner', function() {
         `,
         code: `
-          fn async_thingo() {
-            let (tx, rx) = channel();
-            thread::spawn(move|| {
-                tx.send(10).unwrap();
-            });
-          }
-    
-          fn unused_func() {
-            println!("Never called");
-          }
-    
-          fn broken_func() {
-            println!("This is broken...";
-          }
-    
           fn doubler(n: i32) -> i32 {
             n * 2
           }
         `,
         fixture: `
-        #[test]
-        fn doubler_works() {
-          assert_eq!(doubler(2),4);
-        }
+          #[test]
+          fn doubler_failure() {
+            assert_eq!(doubler(2),3);
+          }
           `,
         testFramework: 'rust'
       }, function(buffer) {
-        expect(buffer.stderr).to.contain(`incorrect close delimiter`);
+        expect(buffer.stdout).to.contain(`<IT::>doubler_failure`);
+        expect(buffer.stdout).to.contain(`<FAILED::>Test Failed`);
         done();
       });
     });
 
-    it('should ignore unused code warnings', function(done) {
+    it('should handle success and failed tests', function(done) {
       runner.run({
         language: 'rust',
-        setup: `
-          use std::thread;
-          use std::sync::mpsc::channel;
-        `,
         code: `
-          fn async_thingo() {
-            let (tx, rx) = channel();
-            thread::spawn(move|| {
-                tx.send(10).unwrap();
-            });
-          }
-    
-          fn unused_func() {
-            println!("Never called");
-          }
-    
           fn doubler(n: i32) -> i32 {
             n * 2
           }
         `,
         fixture: `
-        #[test]
-        fn doubler_works() {
-          assert_eq!(doubler(2),4);
-        }
-          `,
+          #[test]
+          fn doubler_success() {
+            assert_eq!(doubler(2),4);
+          }
+
+          #[test]
+          fn doubler_failure() {
+            assert_eq!(doubler(2),3);
+          }
+        `,
         testFramework: 'rust'
       }, function(buffer) {
-        expect(buffer.stdout).to.contain(`<DESCRIBE::>doubler_works`);
+        expect(buffer.stdout).to.contain(`<IT::>doubler_success`);
         expect(buffer.stdout).to.contain(`<PASSED::>Test Passed`);
+        expect(buffer.stdout).to.contain(`<IT::>doubler_failure`);
+        expect(buffer.stdout).to.contain(`<FAILED::>Test Failed`);
         done();
       });
     });
 
-    it('should handle failed tests', function(done) {
+    it('can use rand', function(done) {
       runner.run({
         language: 'rust',
-        setup: `
-          use std::thread;
-          use std::sync::mpsc::channel;
+        code: 'fn add(x: i32, y: i32) -> i32 { x + y }',
+        // need `self::` because `extern crate` loads to current namespace and tests are in `mod tests`.
+        fixture: `
+          extern crate rand;
+          use self::rand::Rng;
+          use self::rand::distributions::{IndependentSample, Range};
```

Comment:
At a glance, it seems unfortunately that we must use this. Would be good to find a way without it before it goes in since once its in will be kinda locked I guess? Unfortunately I can't suggest an alternative just at a glance as I'm not proficient at rust but am happy to tinker a little bit later (Melbourne time :wink:) to see if there is any help I can provide.